### PR TITLE
TG DNA update attempt #3: post-TGUI preferences edition

### PR DESCRIPTION
**Pull request by Chubbygummibear**
previously https://github.com/yogstation13/Yogstation/pull/14861
# Round 3?

yeah i'm like ~~90%~~ 70% certain i ironed out the preference savefile quirks and it won't break your characters into grayscale monstrosities

Your prefs will auto save with the full hexcode colors while retaining your color choices next time you launch and this is active

# Document the changes in your pull request

More specifically ports

- https://github.com/tgstation/tgstation/pull/59636
- https://github.com/tgstation/tgstation/pull/62303
- https://github.com/tgstation/tgstation/pull/61980 

# What does this actually do?
well if i did everything perfectly right, 90% of players won't notice anything changed. xeno traits such as lizard frills, pod hair, and moth wings are all stored in a separate chunk of dna that everyone has, meaning they can be edited through genetics as well as dna injectors working on the same species instead of only working for and on humans.

# Did you spend a ~~week~~ ~~month~~ year porting tg's dna code just to add stuff to genetics?
yeah

# Wiki Documentation

I'm unsure how much the wiki covers about dna, but maybe a blurb about DNA storing mutant features now, and genetics being able to alter these

# Changelog

:cl:  @Y0SH1M4S73R, @Ghommie, Chubbygummibear
rscadd: The DNA console can now edit the nonhuman features of its occupants, as well as allowing injectors to apply those features. Look for the "Features" tab in the DNA console.
rscadd: everyone has a whole new dna block called unique_features that holds things like mutant color and alien bits settings. unique features can be altered and copied onto other people through genetics. this only works on people of the same species 
wip: this is probably going to need a lot of ongoing work  
tweak: hexcolors are handled differently 
bugfix: glowy color works
bugfix: extendo arm color matches your mcolor
tweak: removes color_legacy prefs and changes sanitize_hexcolor()'s default to 6 characters
experimental: Whole lotta DNA and color code reworking
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
The manmade horrors beyond our comprehension

File: code/__DEFINES/DNA.dm [hunk 1]
```diff
@@ -90,6 +93,30 @@
...
+#define DNA_FEATURE_BLOCKS 19
+
+#define DNA_MUTANT_COLOR_BLOCK 1
+#define DNA_ETHEREAL_COLOR_BLOCK 2
+#define DNA_LIZARD_MARKINGS_BLOCK 3
+#define DNA_LIZARD_TAIL_BLOCK 4
+#define DNA_SNOUT_BLOCK 5
+#define DNA_HORNS_BLOCK 6
+#define DNA_FRILLS_BLOCK 7
+#define DNA_SPINES_BLOCK 8
```
Comment (Moltijoe) [inline code/__DEFINES/DNA.dm:106]: why do these not specify what species they are for?

**Review by Moltijoe — COMMENTED** [review]
a lot of this is just changing variable names and extending colours into proper hex codes

also, color conflicts

File: code/__DEFINES/DNA.dm (hunk 1 above)
Comment (Moltijoe) [reply code/__DEFINES/DNA.dm:106]: also, color conflicts

**Review by Moltijoe — APPROVED** [review]
i can't find anything wrong, most of this is just rearranging code and changing hexvalues
the new parts don't look broken, so hopefully there isn't a secret bug mixed in
perhaps overly hopeful considering this game

**Comment by JohnFulpWillard** [comment]
would be awesome to see this become real

**Comment by Bibby0110** [comment]
Damn this is stale

**Comment by iamthesilly** [comment]
I'll reopen this if the merge conflicts get fixed

**Comment by Chubbygummibear** [comment]
merge conflicts resolved. again. for the morbillionth time. if i see merge conflicts again i would rather delete this entire branch than maintain it again

**Review by ToasterBiome — APPROVED** [review]
looks good

**Pull request merged**